ResearchKit/Common/ORKAnswerFormat_Internal.h

```diff
@@ -135,6 +135,10 @@ ORK_DESIGNATE_CODING_AND_SERIALIZATION_INITIALIZERS(ORKTimeIntervalAnswerFormat)
 - (UIImage *)minimumImage;
 - (NSArray<ORKTextChoice *> *)textChoices;
 
+@optional
```

Comment:
Originally I did it in the style you suggested , but `ORKScaleSliderView` can only accept one type of `format Provider`, so I keep it in one protocol and mark them `optional`.